paasta_tools/docker_hostname_wrapper.py

```diff
@@ -0,0 +1,99 @@
+#!/usr/bin/env python
+""" Meant to be used by mesos-slave instead of the /usr/bin/docker executable
+directly This will parse the CLI arguments intended for docker, extract
+environment variable settings related to the actual node hostname and mesos
+task ID, and use those as an additional --hostname argument when calling the
+underlying docker command.
+
+If the environment variables are unspecified, or if --hostname is already
+specified, this does not change any arguments and just directly calls docker
+as-is.
+"""
+from __future__ import absolute_import
+from __future__ import unicode_literals
+
+import os
+import re
+import socket
+import sys
+
+
+ENV_MATCH_RE = re.compile('^(-\w*e\w*|--env)(=(\S.*))?$')
+MAX_HOSTNAME_LENGTH = 63
+
+
+def parse_env_args(args):
+    result = {}
+    in_env = False
+    for arg in args:
+        if not in_env:
+            match = ENV_MATCH_RE.match(arg)
+            if not match:
+                continue
+            arg = match.group(3) or ''
+            if not arg:
+                in_env = True
+                continue
+
+        in_env = False
+        if '=' not in arg:
+            continue
+
+        k, _, v = arg.partition('=')
+        result[k] = v
+
+    return result
+
+
+def already_has_hostname(args):
+    for arg in args:
+        if arg == '-h':
+            return True
+        if arg.startswith('--hostname'):
+            return True
+        if len(arg) > 1 and arg[0] == '-' and arg[1] != '-':
+            # several short args
+            arg = arg.partition('=')[0]
+            if 'h' in arg:
+                return True
+    return False
+
+
+def generate_hostname(fqdn, mesos_task_id):
+    host_hostname = fqdn.partition('.')[0]
+    task_id = mesos_task_id.rpartition('.')[2]
+
+    hostname = host_hostname + '-' + task_id
+
+    # hostnames can only contain alphanumerics and dashes and must be no more
+    # than 63 characters
+    hostname = re.sub('[^a-zA-Z0-9-]+', '-', hostname)[:MAX_HOSTNAME_LENGTH]
+    return hostname
+
+
+def add_hostname(args, hostname):
+    # Add --hostname argument immediately after 'run' command if it exists
+    args = list(args)
+
+    try:
+        run_index = args.index('run')
+    except ValueError:
+        pass
+    else:
+        args.insert(run_index + 1, '--hostname=' + hostname)
+
+    return args
+
+
+def main(argv=None):
+    argv = argv if argv is not None else sys.argv[1:]
+
+    env_args = parse_env_args(argv)
+    fqdn = socket.getfqdn()
+    mesos_task_id = env_args.get('MESOS_TASK_ID')
```

Comment:
I'm sorry to say this, but this is a marathon specific variable (I know, silly). AFAIK there is no mesos-provided var for this that is universal.

In chronos is is the same thing but *lowercase* `mesos_task_id` :(
https://github.com/yelp/chronos/blob/master/src/main/scala/org/apache/mesos/chronos/scheduler/mesos/MesosTaskBuilder.scala#L57

It looks like upstream head they add both now. We can add that into our fork, but for now can you look for either?